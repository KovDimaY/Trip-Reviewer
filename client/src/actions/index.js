import axios from 'axios';

import {
    GET_TRIPS,
    GET_TRIP_W_REVIEWER,
    CLEAR_TRIP_W_REVIEWER,
    USER_LOGIN,
    USER_AUTH,
    ADD_TRIP,
    CLEAR_NEW_TRIP,
    GET_USER_REVIEWS
} from '../constants/action-names';

export function getTrips(
    limit = 10,
    start = 0,
    order = 'asc',
    list = []
) {
    const request = axios
        .get(`/api/getManyTrips?limit=${limit}&skip=${start}&order=${order}`)
        .then(response => [...list, ...response.data]);

    return {
        type: GET_TRIPS,
        payload: request
    };
}

export function getTripWithReviewer(id) {
    const request = axios.get(`/api/getTrip?id=${id}`);

    return (dispatch) => {
        request.then(({ data }) => {
            const trip = data;

            axios
                .get(`/api/getReviewer?id=${trip.ownerId}`)
                .then(({ data }) => {
                    const response = {
                        trip,
                        reviewer: data
                    };

                    dispatch({
                        type: GET_TRIP_W_REVIEWER,
                        payload: response
                    });
                });
        });
    };
}

export function getUserReviews(userId) {
    const request = axios
        .get(`/api/getUserReviews?user=${userId}`)
        .then(response => response.data);

    return {
        type: GET_USER_REVIEWS,
        payload: request
    };
}

export function clearTripWithReviewer() {
    return {
        type: CLEAR_TRIP_W_REVIEWER,
        payload: {
            trip: null,
            reviewer: null
        }
    };
}

export function loginUser({ email, password }) {
    const request = axios
        .post('/api/login',{ email, password })
        .then(response => response.data);

    return {
        type: USER_LOGIN,
        payload: request
    };
}

export function auth() {
    const request = axios
        .get('/api/auth')
        .then(response => response.data);

    return {
        type: USER_AUTH,
        payload: request
    }
}

export function addTrip(trip) {
    const request = axios
        .post('/api/trip', trip)
        .then(response => response.data);

    return {
        type: ADD_TRIP,
        payload: request
    };
}

export function clearNewTrip() {
    return {
        type: CLEAR_NEW_TRIP,
        payload: null
    };
}