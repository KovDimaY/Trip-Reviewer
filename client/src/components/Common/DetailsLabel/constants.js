export const TYPES = {
  expences: 'EXPENCES',
  duration: 'DURATION',
  rating: 'RATING',
};

const colors = {
  very_good: '#00B300',
  good: '#7FBB17',
  ok: '#FEC22D',
  bad: '#F47048',
  very_bad: '#E91E63',
};

const limitsExpences = [100, 500, 1000, 5000];
const limitsDuration = [2, 7, 14, 31];
const limitsRating = [1, 2, 3, 4];

const getColorsInversed = (value, limits) => {
  switch (true) {
    case value > limits[3]:
      return colors.very_bad;
    case value > limits[2]:
      return colors.bad;
    case value > limits[1]:
      return colors.ok;
    case value > limits[0]:
      return colors.good;
    default:
      return colors.very_good;
  }
};

const getColorsDirect = (value, limits) => {
  switch (true) {
    case value > limits[3]:
      return colors.very_good;
    case value > limits[2]:
      return colors.good;
    case value > limits[1]:
      return colors.ok;
    case value > limits[0]:
      return colors.bad;
    default:
      return colors.very_bad;
  }
};

const getFunctionByType = {
  [TYPES.expences]: value => getColorsInversed(value, limitsExpences),
  [TYPES.duration]: value => getColorsDirect(value, limitsDuration),
  [TYPES.rating]: value => getColorsDirect(value, limitsRating),
};

export const getColorsByTypeAndValue = (type, value) => getFunctionByType[type](value);
