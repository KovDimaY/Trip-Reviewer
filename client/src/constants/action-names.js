export const GET_TRIPS = 'get-trips';
export const GET_TRIP_W_REVIEWER = 'get-trip-w-reviewer';
export const CLEAR_TRIP_W_REVIEWER = 'clear-trip-w-reviewer';
export const USER_LOGIN = 'user-login';
export const USER_AUTH = 'user-auth';