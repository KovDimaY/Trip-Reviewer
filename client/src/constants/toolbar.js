const toolbar = {
  options: ['inline', 'list', 'textAlign', 'link', 'emoji', 'image', 'remove', 'history'],
  inline: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['bold', 'italic', 'underline', 'strikethrough'],
  },
  list: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['unordered', 'ordered', 'indent', 'outdent'],
  },
  textAlign: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['left', 'center', 'right', 'justify'],
  },
  link: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    dropdownClassName: undefined,
    showOpenOptionOnHover: true,
    defaultTargetOption: '_self',
  },
  emoji: {
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    emojis: [
      '😀', '😁', '😂', '😃', '😉', '😋', '😎', '😍', '😗', '🤗', '🤔', '😣', '😫', '😴', '😌', '🤓',
      '😛', '😜', '😠', '😇', '😷', '😈', '👻', '😺', '😸', '😹', '😻', '😼', '😽', '🙀', '🙈',
      '🙉', '🙊', '👼', '👮', '🕵', '💂', '👳', '🎅', '👸', '👰', '👲', '🙍', '🙇', '🚶', '🏃', '💃',
      '⛷', '🏂', '🏌', '🏄', '🚣', '🏊', '⛹', '🏋', '🚴', '👫', '💪', '👈', '👉', '👉', '👆', '🖕',
      '👇', '🖖', '🤘', '🖐', '👌', '👍', '👎', '✊', '👊', '👏', '🙌', '🙏', '🐵', '🐶', '🐇', '🐥',
      '🐸', '🐌', '🐛', '🐜', '🐝', '🍉', '🍄', '🍔', '🍤', '🍨', '🍪', '🎂', '🍰', '🍾', '🍷', '🍸',
      '🍺', '🌍', '🚑', '⏰', '🌙', '🌝', '🌞', '⭐', '🌟', '🌠', '🌨', '🌩', '⛄', '🔥', '🎄', '🎈',
      '🎉', '🎊', '🎁', '🎗', '🏀', '🏈', '🎲', '🔇', '🔈', '📣', '🔔', '🎵', '🎷', '💰', '🖊', '📅',
      '✅', '❎', '💯',
    ],
  },
  image: {
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    urlEnabled: true,
    uploadEnabled: true,
    alignmentEnabled: true,
    uploadCallback: undefined,
    previewImage: false,
    inputAccept: 'image/gif,image/jpeg,image/jpg,image/png,image/svg',
    alt: { present: false, mandatory: false },
    defaultSize: {
      height: 'auto',
      width: 'auto',
    },
  },
  remove: { className: undefined, component: undefined },
  history: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['undo', 'redo'],
  },
};

export default toolbar;
