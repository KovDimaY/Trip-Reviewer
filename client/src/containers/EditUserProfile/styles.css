.edit-user-profile-container {
    max-width: 600px;
    margin: 0 auto 75px auto;
}
.edit-user-profile-container .avatar {
    text-align: center;
    margin-top: 30px;
}
.edit-user-profile-container .info {
    box-sizing: border-box;
    padding:20px;
}
.edit-user-profile-container .info .form_element {
    font-size: 23px;
    font-weight: 300;
    margin-top: 10px;
    color: #a4a4a4;
}
.edit-user-profile-container .info div span {
    display: block;
    font-weight: 500;
    color: #02bcd4;
}
.edit-user-profile-container .danger div span {
    display: block;
    color: orange;
}
.edit-user-profile-container .danger-zone {
    color: orangered;
    text-align: center;
    font-size: 24px;
    font-weight: bold;
    margin-bottom: 0;
}
.edit-user-profile-container .disclamer {
    color: darkgray;
    text-align: center;
    font-size: 12px;
    margin-bottom: -5px;
}
.edit-user-profile-container .disclamer-error {
    color: red;
    text-align: center;
    font-size: 14px;
    margin-bottom: -5px;
}
.edit-user-profile-container .text-center {
    margin-bottom: 30px;
}
.edit-user-profile-container .info .margin-top {
    margin-top: 50px;
}
.edit-user-profile-container input {
    width: -webkit-fill-available;
    border: solid 1px gray;
    height: 36px;
    margin: 5px 0;
    padding-left: 10px;
    font-size: 16px;
}
.edit-user-profile-container .user-avatar-controls {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    padding: 20px;
}
.edit-user-profile-container .user-avatar-controls .avatar-button {
    text-align: center;
    display: block;
    color: white;
    padding: 10px;
    border-radius: 4px;
    width: 35%;
}
.edit-user-profile-container .user-avatar-controls .update-user-avatar {
    background-color: #02bcd4;
}
.edit-user-profile-container .user-avatar-controls .delete-user-avatar {
    background-color: orange;
}
.edit-user-profile-container .button-link {
    font-size: 20px;
    padding: 10px 52px;
    background: #02bcd4;
    color: #fff;
    font-weight: 300;
    border-radius: 3px;
    border: none;
    margin-top: 20px;
    width: 260px;
}
.edit-user-profile-container .calcel {
    background-color: orange;
}
.edit-user-profile-container .error {
    color: #FF5722;
    margin-top: 5px;
    text-align: center;
}
.edit-user-profile-container .field-error {
    border: 2px solid #FF5722;
    color: #FF5722;
}
.edit-user-profile-container .avatar {
    width: 200px;
    height: 200px;
    margin: 50px auto 25px auto;
}