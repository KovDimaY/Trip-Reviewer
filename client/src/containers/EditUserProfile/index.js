import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Progress } from 'react-sweet-progress';

import ImageUploader from './../../components/ImageUploader';
import UserAvatar from './../../components/UserAvatar';
import { updateUser } from './../../actions';
import { firebase } from './../../firebase';

import './styles.css';
import "react-sweet-progress/lib/style.css";

class EditUserProfile extends PureComponent {
    state = {
        formdata: {
            _id: this.props.users.login.id,
            name: this.props.users.login.name,
            lastname: this.props.users.login.lastname,
            email: this.props.users.login.email,
            avatar: this.props.users.login.avatar,
            oldPassword: '',
            newPassword: '',
            repeatPassword: ''
        },
        isUploading: false,
        progress: 0,
        uploadError: null
    };

    componentWillMount() {
        firebase.auth().signInAnonymously().catch((error) => {
            console.log(error);
        });
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.result && nextProps.result.success) {
            nextProps.history.push('/user');
        }
    }

    submitForm = (event) => {
        event.preventDefault();
        this.props.dispatch(updateUser(this.state.formdata));
    }

    handleInput = (event) => {
        const newFormdata = {
            ...this.state.formdata
        };
        const { value, name } = event.target;

        newFormdata[name] = value;

        this.setState({ formdata: newFormdata });
    }

    handleDeleteAvatar = () => {
        const newFormdata = {
            ...this.state.formdata
        };

        newFormdata.avatar = null;

        this.setState({ formdata: newFormdata });
    }

    onUploadStarts = () => {
        this.setState({
            isUploading: true,
            progress: 0,
            uploadError: null
        });
    }

    onUploadProgress = (progress) => {
        this.setState({ progress });
    }

    onUploadSuccess = (filename) => {
        const newFormdata = {
            ...this.state.formdata
        };

        newFormdata.avatar = filename;

        this.setState({
            formdata: newFormdata,
            isUploading: false,
            progress: 100
        });
    }

    onUploadError = (error) => {
        this.setState({ isUploading: false, uploadError: error });
    }

    renderAvatarControls() {
        const { isUploading, progress } = this.state;
        
        if (isUploading) {
            return (
                <div className="user-avatar-controls">
                    <Progress percent={progress} />
                </div>
            );
        }
        return (
            <div className="user-avatar-controls">
                <ImageUploader
                    filename={this.props.users.login.id}
                    onUploadStarts={this.onUploadStarts}
                    onUploadProgress={this.onUploadProgress}
                    onUploadSuccess={this.onUploadSuccess}
                    onUploadError={this.onUploadError}
                    className="avatar-button update-user-avatar"
                />
                <label className="avatar-button delete-user-avatar" onClick={this.handleDeleteAvatar}>Delete avatar</label>
            </div>
        );
    }

    renderSoftInfoInputs() {
        const { name, lastname } = this.state.formdata;

        return (
            <div className="info">
                <div className="form_element">
                    <span>Name:</span>
                    <input
                        type="text"
                        name="name"
                        placeholder="Enter name"
                        value={name}
                        onChange={this.handleInput}
                    />
                </div>

                <div className="form_element">
                    <span>Lastname:</span>
                    <input
                        type="text"
                        name="lastname"
                        placeholder="Enter lastname"
                        value={lastname}
                        onChange={this.handleInput}
                    />
                </div>
            </div>
        );
    }

    renderDangerZoneInputs() {
        const {
            oldPassword, newPassword,
            repeatPassword, email
        } = this.state.formdata;

        return (
            <div className="info danger">
                <div className="form_element current-password">
                    <span>Current Password:</span>
                    <input
                        type="password"
                        name="oldPassword"
                        placeholder="Enter your current password"
                        value={oldPassword}
                        onChange={this.handleInput}
                    />
                </div>

                <div className="form_element">
                    <span>New Password:</span>
                    <input
                        type="password"
                        name="newPassword"
                        placeholder="Enter your new password"
                        value={newPassword}
                        onChange={this.handleInput}
                    />
                </div>

                <div className="form_element">
                    <span>Repeat New Password:</span>
                    <input
                        type="password"
                        name="repeatPassword"
                        placeholder="Repeat your new password"
                        value={repeatPassword}
                        onChange={this.handleInput}
                    />
                </div>

                <div className="form_element">
                    <span>Email:</span>
                    <input
                        type="email"
                        name="email"
                        placeholder="Enter new email"
                        value={email}
                        onChange={this.handleInput}
                    />
                </div>
            </div>
        );
    }

    renderDisclaimer() {
        if (this.state.uploadError) {
            return (
                <p className="disclamer-error">
                    Upload was not successful.<br/>
                    The avatar image should be smaller than <b>1MB</b>.
                </p>
            );
        }
        return (
            <p className="disclamer">
                * To be sure that your avatar will be updated correctly <br/>
                press <b>"Submit changes"</b> button at the bottom of the page
            </p>
        )
    }

    renderError() {
        const { result } = this.props;

        if (result && result.message) {
            return (
                <div className="error">
                    { this.props.result.message }
                </div>
            );
        }
    }

    render() {
        const { avatar } = this.state.formdata;

        return (
            <div className="edit-user-profile-container">
                <UserAvatar filename={avatar} />

                { this.renderDisclaimer() }
                { this.renderAvatarControls() }

                <form onSubmit={this.submitForm}>
                    { this.renderSoftInfoInputs() }

                    <hr />
                    <p className="danger-zone">Danger zone</p>

                    { this.renderDangerZoneInputs() }

                    { this.renderError() }

                    <div className="text-center">
                        <button type="submit" className="button-link">Submit changes</button>
                    </div>
                </form>
                <div className="text-center">
                    <Link to="/user" className="button-link calcel">Cancel</Link>
                </div>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        result: state.users.userUpdate
    }
}

export default connect(mapStateToProps)(EditUserProfile);