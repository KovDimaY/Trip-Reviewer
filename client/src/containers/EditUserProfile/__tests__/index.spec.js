import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
 
import EditUserProfile from './../../EditUserProfile';
import { UPDATE_USER } from './../../../constants/action-names';
import { updateUser } from './../../../actions';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));
jest.mock('./../../../actions', () => ({ 
    updateUser: jest.fn(() => ({
        type: UPDATE_USER,
        payload: {}
    })),
}));

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
    const store = mockStore(initialState);

    return (
        <EditUserProfile {...props} store={store} />
    );
};
 
describe('<EditUserProfile />', () => {
    it('should render component', () => {
        const initialState = {
            users: {}
        };
        const props = {
            users: {
                login: {
                    id: 'id'
                }
            }
        };
        const tree = create(mockComponent(initialState, props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('should render component with error', () => {
        const initialState = {
            users: {
                userUpdate: {
                    success: false,
                    message: 'test'
                },
            }
        };
        const props = {
            users: {
                login: {
                    id: 'id'
                }
            }
        };
        const tree = create(mockComponent(initialState, props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('should redirect when componentWillReceiveProps is called with success true', () => {
        const push = jest.fn();
        const path = '/user';
        const initialState = {
            users: {
                userUpdate: {},
            }
        };
        const props = {
            users: {
                login: {
                    id: 'id'
                }
            }
        };
        const nextProps = {
            result: {
                success: true
            },
            history: {
                push
            }
        };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();
    
        instance.componentWillReceiveProps(nextProps);
    
        expect(push).toHaveBeenCalledWith(path);
    });

    it('should not redirect when componentWillReceiveProps is called with success false', () => {
        const push = jest.fn();
        const path = '/user';
        const initialState = {
            users: {
                userUpdate: {},
            }
        };
        const props = {
            users: {
                login: {
                    id: 'id'
                }
            }
        };
        const nextProps = {
            result: {
                success: false,
                message: 'test'
            },
            history: {
                push
            }
        };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();
    
        instance.componentWillReceiveProps(nextProps);
    
        expect(push).not.toHaveBeenCalledWith(path);
    });

    it('should dispatch updateUser when submitForm is called', () => {
        const preventDefault = jest.fn();
        const initialState = {
            users: {
                userUpdate: {},
            }
        };
        const props = {
            users: {
                login: {
                    id: 'id'
                }
            }
        };
        const state = {
            name: "name",
            lastname: "lastname",
            email: "email"
        };
        const event = { preventDefault };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();
        instance.setState({ formdata: state });
    
        instance.submitForm(event);

        expect(preventDefault).toHaveBeenCalled();    
        expect(updateUser).toHaveBeenCalledWith(state);
    });

    it('handleInput should change state correctly', () => {
        const initialState = {
            users: {
                userUpdate: {},
            }
        };
        const props = {
            users: {
                login: {
                    id: 'id'
                }
            }
        };
        const event = {
            target: {
                value: 'test',
                name: 'title'
            }
        };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();
    
        instance.handleInput(event);
    
        expect(instance.state.formdata.title).toEqual('test');
    });
});