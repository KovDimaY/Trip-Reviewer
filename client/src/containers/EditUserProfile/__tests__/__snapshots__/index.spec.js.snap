// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditUserProfile /> should render component 1`] = `
<div
  className="edit-user-profile-container"
>
  <div
    className="avatar"
  >
    <UserAvatar
      filename={undefined}
    />
  </div>
  <p
    className="disclamer"
  >
    * To be sure that your avatar will be updated correctly
     
    <br />
    press
     
    <b>
      "
      Submit changes
      "
    </b>
     
    button at the bottom of the page
  </p>
  <div
    className="user-avatar-controls"
  >
    <ImageUploader
      className="avatar-button update-user-avatar"
      filename="id"
      onUploadError={[Function]}
      onUploadProgress={[Function]}
      onUploadStarts={[Function]}
      onUploadSuccess={[Function]}
    />
    <label
      className="avatar-button delete-user-avatar"
      onClick={[Function]}
    >
      Delete avatar
    </label>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="info"
    >
      <div
        className="form-element"
      >
        <span>
          Name:
        </span>
        <input
          className=""
          name="name"
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
          value={undefined}
        />
      </div>
      <div
        className="form-element"
      >
        <span>
          Lastname:
        </span>
        <input
          className=""
          name="lastname"
          onChange={[Function]}
          placeholder="Enter lastname"
          type="text"
          value={undefined}
        />
      </div>
    </div>
    <hr />
    <p
      className="danger-zone"
    >
      Danger zone
    </p>
    <div
      className="info danger"
    >
      <div
        className="form-element"
      >
        <span>
          Current Password:
        </span>
        <input
          className=""
          name="oldPassword"
          onChange={[Function]}
          placeholder="Enter your current password"
          type="password"
          value=""
        />
      </div>
      <div
        className="form-element margin-top"
      >
        <span>
          New Password:
        </span>
        <input
          className=""
          name="newPassword"
          onChange={[Function]}
          placeholder="Enter your new password"
          type="password"
          value=""
        />
      </div>
      <div
        className="form-element"
      >
        <span>
          Repeat New Password:
        </span>
        <input
          className=""
          name="repeatPassword"
          onChange={[Function]}
          placeholder="Repeat your new password"
          type="password"
          value=""
        />
      </div>
      <div
        className="form-element"
      >
        <span>
          Email:
        </span>
        <input
          className=""
          name="email"
          onChange={[Function]}
          placeholder="Enter new email"
          type="email"
          value={undefined}
        />
      </div>
    </div>
    <div
      className="text-center"
    >
      <button
        className="button-link"
        type="submit"
      >
        Submit changes
      </button>
    </div>
  </form>
  <div
    className="text-center"
  >
    <Link
      className="button-link calcel"
      to="/user"
    >
      Cancel
    </Link>
  </div>
</div>
`;

exports[`<EditUserProfile /> should render component with error 1`] = `
<div
  className="edit-user-profile-container"
>
  <div
    className="avatar"
  >
    <UserAvatar
      filename={undefined}
    />
  </div>
  <p
    className="disclamer"
  >
    * To be sure that your avatar will be updated correctly
     
    <br />
    press
     
    <b>
      "
      Submit changes
      "
    </b>
     
    button at the bottom of the page
  </p>
  <div
    className="user-avatar-controls"
  >
    <ImageUploader
      className="avatar-button update-user-avatar"
      filename="id"
      onUploadError={[Function]}
      onUploadProgress={[Function]}
      onUploadStarts={[Function]}
      onUploadSuccess={[Function]}
    />
    <label
      className="avatar-button delete-user-avatar"
      onClick={[Function]}
    >
      Delete avatar
    </label>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="info"
    >
      <div
        className="form-element"
      >
        <span>
          Name:
        </span>
        <input
          className="field-error"
          name="name"
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
          value={undefined}
        />
      </div>
      <div
        className="error"
      >
        test
      </div>
      <div
        className="form-element"
      >
        <span>
          Lastname:
        </span>
        <input
          className="field-error"
          name="lastname"
          onChange={[Function]}
          placeholder="Enter lastname"
          type="text"
          value={undefined}
        />
      </div>
      <div
        className="error"
      >
        test
      </div>
    </div>
    <hr />
    <p
      className="danger-zone"
    >
      Danger zone
    </p>
    <div
      className="info danger"
    >
      <div
        className="form-element"
      >
        <span>
          Current Password:
        </span>
        <input
          className="field-error"
          name="oldPassword"
          onChange={[Function]}
          placeholder="Enter your current password"
          type="password"
          value=""
        />
      </div>
      <div
        className="error"
      >
        test
      </div>
      <div
        className="form-element margin-top"
      >
        <span>
          New Password:
        </span>
        <input
          className="field-error"
          name="newPassword"
          onChange={[Function]}
          placeholder="Enter your new password"
          type="password"
          value=""
        />
      </div>
      <div
        className="error"
      >
        test
      </div>
      <div
        className="form-element"
      >
        <span>
          Repeat New Password:
        </span>
        <input
          className="field-error"
          name="repeatPassword"
          onChange={[Function]}
          placeholder="Repeat your new password"
          type="password"
          value=""
        />
      </div>
      <div
        className="error"
      >
        test
      </div>
      <div
        className="form-element"
      >
        <span>
          Email:
        </span>
        <input
          className="field-error"
          name="email"
          onChange={[Function]}
          placeholder="Enter new email"
          type="email"
          value={undefined}
        />
      </div>
      <div
        className="error"
      >
        test
      </div>
    </div>
    <div
      className="text-center"
    >
      <button
        className="button-link"
        type="submit"
      >
        Submit changes
      </button>
    </div>
  </form>
  <div
    className="text-center"
  >
    <Link
      className="button-link calcel"
      to="/user"
    >
      Cancel
    </Link>
  </div>
</div>
`;
